packages/thirdweb/src/react/native/ui/components/RNImage.tsx

```diff
@@ -31,27 +34,34 @@ function getImage(data: string): {
 
 export const RNImage = (props: ImageInfo) => {
   const { data, size, color, placeholder } = props;
+  const [loading, setLoading] = useState(true);
   if (!data) {
     return null;
   }
   const { image, type } = useMemo(() => getImage(data), [data]);
   switch (type) {
     case "url":
-      return (
-        <Image
-          source={{ uri: image }}
-          loadingIndicatorSource={{ uri: placeholder }}
-          width={size}
-          height={size}
-          style={{ borderRadius: radius.md }}
-        />
-      );
     case "image":
       return (
-        <Image
-          source={{ uri: image }}
-          style={{ width: size, height: size, borderRadius: radius.md }}
-        />
+        <>
+          {loading && (
+            <Skeleton
```

Comment:
I wonder if this "absolutely positioned" thing scales for skeletons?

it makes sense that you want to render `Image` since it has the loading indicator on it... maybe skeleton could take `Image` as a child and render it (and internally optionally overlay the skeleton animation? That way it should (?) be re-usable for other places that need a skeleton ui?